### PR TITLE
fix: Typography mixin arguments & update docs

### PR DESCRIPTION
**Pull request by starryeyez024**
## <Pull request description>

- Add support to the $base argument so that it works regardless if the $type argument is "title" or "text". Right now it does not work if the $type is title.
- Updates pfe-clipboard component so that it leverages system typography variables
- Add alias mixin for easier use within components:  `@mixin pfe-c-typography`

### Related issues

- https://github.com/patternfly/patternfly-elements/issues/1384

### Preview
 
Link(s) to demo page(s) where this element can be viewed:
- [pfe-styles](https://deploy-preview-1385--happy-galileo-ea79c4.netlify.app/elements/pfe-styles/demo/)
- [pfe-clipboard](https://deploy-preview-1385--happy-galileo-ea79c4.netlify.app/elements/pfe-clipboard/demo/)
- [pfe-sass docs](https://deploy-preview-1385--happy-galileo-ea79c4.netlify.app/elements/pfe-sass/demo/#typography-mixin-pfe-typography)

### Sandbox links

 - https://web-dev-drupal-webux3.dev.a1.vary.redhat.com/en/demo-web-components#classes
 - https://web-dev-drupal-webux3.dev.a1.vary.redhat.com/en/potter-long-form-article-test

### What has changed and why

- Modified pfe-typography mixin to universally apply $base argument to title & text types
- Added alias mixin to make it easier for components to just get the font-size they need

### Testing instructions

Nothing should regress, because the PFE project is not setting base to false (its more useful for other projects like webrh). You can verify a few pages like [pfe-band](https://deploy-preview-1385--happy-galileo-ea79c4.netlify.app/elements/pfe-band/demo/) - they should look the same

If you are using the `pfe-typography` function, you should be able to opt out of the extra properties (beyond font-family, font-size) if desired, via the `$base` argument set to false:

Test by throwing this sass into some .scss file in pfe-styles:

```scss
.foo {
  @extend %pfe-title--lg;
}
.bar {
 @include pfe-typography(lg, $type: "title"); 
}
.baz {
 @include pfe-typography(lg, $type: "title", $base: false); 
}
```

the expected output should return:

```css
.foo {
  font-family: "Red Hat Display", "RedHatDisplay", "Overpass", Overpass, Arial, sans-serif;
  font-family: var(--pfe-theme--font-family--heading, "Red Hat Display", "RedHatDisplay", "Overpass", Overpass, Arial, sans-serif);
  font-size: 18px;
  font-size: var(--pf-c-title--m-lg--FontSize, var(--pf-global--FontSize--lg, 18px));
  line-height: 1.5;
  line-height: var(--pfe-theme--line-height, 1.5);
  font-weight: 400;
  font-weight: var(--pfe-theme--font-weight--normal, 400);
}

.foo:not(:last-child) {
  margin-bottom: 16px;
  margin-bottom: var(--pfe-theme--content-spacer--heading--sm, 16px);
}

.bar {
  font-family: "Red Hat Display", "RedHatDisplay", "Overpass", Overpass, Arial, sans-serif;
  font-family: var(--pfe-theme--font-family--heading, "Red Hat Display", "RedHatDisplay", "Overpass", Overpass, Arial, sans-serif);
  font-size: 18px;
  font-size: var(--pf-c-title--m-lg--FontSize, var(--pf-global--FontSize--lg, 18px));
  line-height: 1.5;
  line-height: var(--pfe-theme--line-height, 1.5);
  font-weight: 400;
  font-weight: var(--pfe-theme--font-weight--normal, 400);
}

.bar:not(:last-child) {
  margin-bottom: 16px;
  margin-bottom: var(--pfe-theme--content-spacer--heading--sm, 16px);
}

.baz {
  font-family: "Red Hat Display", "RedHatDisplay", "Overpass", Overpass, Arial, sans-serif;
  font-family: var(--pfe-theme--font-family--heading, "Red Hat Display", "RedHatDisplay", "Overpass", Overpass, Arial, sans-serif);
  font-size: 18px;
  font-size: var(--pf-c-title--m-lg--FontSize, var(--pf-global--FontSize--lg, 18px));
  line-height: 1.5;  /* This shouldn't be here */
  line-height: var(--pfe-theme--line-height, 1.5); /* This shouldn't be here */
  font-weight: 400; /* This shouldn't be here */
  font-weight: var(--pfe-theme--font-weight--normal, 400); /* This shouldn't be here */
}

.baz:not(:last-child) {
  margin-bottom: 16px; /* This shouldn't be here */
  margin-bottom: var(--pfe-theme--content-spacer--heading--sm, 16px); /* This shouldn't be here */
}
```
 
## Full list of things you can throw into the mixin

- http://pastebin.test.redhat.com/947129

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one request or issue (no stragglers or scope-creep).
- n/a Tests have been updated to cover these changes.
- n/a Browser testing passed.
- [x] Repository compiles and tests pass.
- [x] Changelog updated.
- [x] Documentation (README.md, WHY.md, etc.) updated or added:  https://github.com/patternfly/patternfly-elements/pull/1395 
- n/a Approved by designer.

### Merging

Please **squash** when merging and ensure your commit message uses [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) formatting.

**Be sure to share your updates with the [patternfly-elements-contribute@redhat.com](mailto:patternfly-elements-contribute@redhat.com) mailing list!**

### CODE REVIEW

File: elements/pfe-cta/src/pfe-cta.scss [hunk 1]
```diff
@@ -25,9 +25,11 @@ $LOCAL-VARIABLES: (
   TextDecoration: none,
   TextDecoration--hover: none,
   TextDecoration--focus: none,
-  FontWeight: pfe-var(font-weight--semi-bold),
+  FontWeight: pfe-var(font-weight--bold),
```
Comment (starryeyez024) [inline elements/pfe-cta/src/pfe-cta.scss:28]: idk why this is diffing. The master branch has line 28 as bold

File: elements/pfe-sass/mixins/_copy-mixins.scss [hunk 2]
```diff
@@ -208,25 +208,25 @@
...
+            font-family: $font-family#{$rules};
             font-size:   $font-size#{$rules};
             @if $base {  // these styles are ordinarily inherited from the body tag, but some projects may need extra specificity
-                font-family: $font-family#{$rules};
                 line-height: $line-height#{$rules};
                 font-weight: $font-weight#{$rules};
-            }
-            &:not(:last-child):not(:empty) {
-                margin-bottom: $margin-bottom#{$rules}; //16px
+                &:not(:last-child):not(:empty) {
```
Comment (starryeyez024) [inline elements/pfe-sass/mixins/_copy-mixins.scss:227]: should the pfe-typography mixin allow you to opt out of margins separately from the other font styles? so if i want font-family, font-size, line-height, and font-weight, I can get those things but avoid the margins

Comment (starryeyez024) [reply elements/pfe-sass/mixins/_copy-mixins.scss:227]: The alternative is you must skip the mixin and call things one by one:
    font-family: pfe-var(font-family--heading);
    font-size: pfe-var(FontSize--xl);
    line-height: pfe-var(line-height);
    font-weight: pfe-var(font-weight--normal);

**Comment by starryeyez024** [comment]
Note: the `$base: false` argument is still printing margins

File: elements/pfe-sass/mixins/_copy-mixins.scss (hunk 2 above)
Comment (castastrophe) [reply elements/pfe-sass/mixins/_copy-mixins.scss:227]: I thought the goal of base was to provide attributes that are generally inherited like font-family and line-height?

Comment from line 224 supports this:
`// these styles are ordinarily inherited from the body tag, but some projects may need extra specificity`

Comment (castastrophe) [reply elements/pfe-sass/mixins/_copy-mixins.scss:227]: Since margin is not inherited, I'm not sure it makes sense to invoke it with base

Comment (starryeyez024) [reply elements/pfe-sass/mixins/_copy-mixins.scss:227]: ++ Potter said the same when we were reviewing , I'm going to break that one out into a new argument

**Comment by starryeyez024** [comment]
@heyMP this has been updated; you could copy/paste the contents of _copy-mixins.scss from here into the file in your branch to test w/ your components

**Review by castastrophe — APPROVED** [review]
Love it!

File: elements/pfe-clipboard/src/pfe-clipboard.scss [hunk 3]
```diff
@@ -5,7 +5,6 @@ $LOCAL: clipboard;
 
 // This variable is global so that helper functions can reference it
 $LOCAL-VARIABLES: (
-  FontSize: 1rem,
```
Comment (castastrophe) [inline elements/pfe-clipboard/src/pfe-clipboard.scss:8]: @starryeyez024 Should we be deprecating functionality?  What if someone is using this already?

File: elements/pfe-sass/mixins/_copy-mixins.scss [hunk 4]
```diff
@@ -250,34 +246,107 @@
...
+/// .jump-links-text {
+///    @include pfe-c-typography($type: text, $sizing: md, $region: header, $light-dom-heading: true)
+/// }
+/// @example - css - Rendered result
+/// .jump-links-text {
+///   font-size: 1rem;
+///   font-size: var(--pfe-clipboard__header--FontSize, var(--pf-global--FontSize--md, 1rem));
+///   --pf-c--FontSize: var(--pfe-clipboard--FontSize, 1rem);
+/// }
+@mixin pfe-c-typography (
```
Comment (castastrophe) [inline elements/pfe-sass/mixins/_copy-mixins.scss:332]: Nice - very PatternFly

File: elements/pfe-styles/README.md [hunk 5]
```diff
@@ -2,39 +2,245 @@
 
 ## Usage
 
-PatternFly Elements Styles is available to provide helper styles that can be directly included on your page.  This current includes css for layouts, context, and our flavor of "normalize".
+PatternFly Elements Styles are available to provide helper styles that can be directly included on your page.  This currently includes CSS for layouts, context, typography, and our flavor of "normalize".
```
Comment (castastrophe) [inline elements/pfe-styles/README.md:5]:
```suggestion
PatternFly Elements Styles are available to provide helper styles that can be directly included on your page.  This currently includes CSS for layouts, context, typography, and our flavour of "normalize".
```

haha totally kidding.  Don't British-ize the docs 😆

**Review by castastrophe — APPROVED** [review]
Loving the updated documentation.  I added a few minor notes but take them if you will 😄

File: elements/pfe-clipboard/src/pfe-clipboard.scss (hunk 3 above)
Comment (starryeyez024) [reply elements/pfe-clipboard/src/pfe-clipboard.scss:8]: this doesn't actually do anything now. previously it was responsible for creating the fallback value in the clipboard CSS, but now the typography mixin does that. The resulting CSS is identical to what it is on master though 👍

**Pull request merged**